js/binance.js

```diff
@@ -1596,23 +1658,27 @@ module.exports = class binance extends Exchange {
          * @param {object} params extra parameters specific to the exchange api endpoint
          * @returns {[object]} an array of objects representing market data
          */
-        const defaultType = this.safeString2 (this.options, 'fetchMarkets', 'defaultType', 'spot');
-        const type = this.safeString (params, 'type', defaultType);
-        const query = this.omit (params, 'type');
-        const spot = (type === 'spot');
-        const margin = (type === 'margin');
-        const future = (type === 'future');
-        const delivery = (type === 'delivery');
-        if ((!spot) && (!margin) && (!future) && (!delivery)) {
-            throw new ExchangeError (this.id + " does not support '" + type + "' type, set exchange.options['defaultType'] to 'spot', 'margin', 'delivery' or 'future'"); // eslint-disable-line quotes
-        }
-        let method = 'publicGetExchangeInfo';
-        if (future) {
-            method = 'fapiPublicGetExchangeInfo';
-        } else if (delivery) {
-            method = 'dapiPublicGetExchangeInfo';
+        let promises = [];
+        const fetchMarkets = this.safeValue (this.options, 'fetchMarkets', [ 'spot', 'linear', 'inverse' ]);
+        for (let i = 0; i < fetchMarkets.length; i++) {
+            const marketType = fetchMarkets[i];
+            if (marketType === 'spot') {
+                promises.push (this.publicGetExchangeInfo (params));
+            } else if (marketType === 'linear') {
+                promises.push (this.fapiPublicGetExchangeInfo (params));
+            } else if (marketType === 'inverse') {
+                promises.push (this.dapiPublicGetExchangeInfo (params));
+            } else {
+                throw new ExchangeError (this.id + ' fetchMarkets() this.options fetchMarkets "' + marketType + '" is not a supported market type');
+            }
         }
-        const response = await this[method] (query);
+        promises = await Promise.all (promises);
+        const spotMarkets = this.safeValue (promises[0], 'symbols', []);
+        const futureMarkets = this.safeValue (promises[1], 'symbols', []);
+        const deliveryMarkets = this.safeValue (promises[2], 'symbols', []);
```

Comment:
this is silently crashing in python because for `binanceusdm/binancecoinm` pomises length is 1, so we get a "list index out of range"

Reply:
Good catch!

Reply:
@kroitor going to to push a fix soon